Feature - Restart Server at Specific Times

DomicidalDesigns opened the pull request:
This adds the feature requested in #89.

**Command Changes**
- Added `ServerRestartScheduleType` config option to choose between interval or time for automatic server restarts
- Added Discord command to change the conf option mentioned above (`!set_restart_schedule_type`)
- Added list of automatic server restart times to the config file 
- Added Discord command to change the list of restart times (`!set_restart_time`)
- Added these restart server config settings to `!get_settings` command

**Scheduler Changes**
- Created `GetIntervalFromTimes()` method to return interval for the next restart time

**BotSettings Changes**
- Added `GetServerRestartSchedule()` method to supply Scheduler with the correct milliseconds interval (time based or interval based), this is based on the `ServerRestartScheduleType` setting being "interval" or "time".

**Localization & README**
- Localized discord command messages this time :)
- Added descriptions for new Discord Bot Commands
- Changed all parameter brackets in README.md from angled to rectangle (angled was not displaying due to Markdown issue).

Note: added `new JsonSerializerSettings{ObjectCreationHandling = ObjectCreationHandling.Replace}` to `JsonConvert.DeserializeObject()` in Program.cs because in my testing the default list of restart times was being added to the config list of times instead of being replaced. This is due to Json serializer default behavior.

egebilecen commented:
I will try to review it today! 😊

egebilecen commented:
Hey, I been quite busy so I haven't had chance to review it yet, sorry about it. I will get try to review the PR asap. Also, there doesn't seem to be `default.json` among edited files. Did you run the project after adding records to the localization dictionary? Because it should have exported new records to the said file (and git should have catched it).

DomicidalDesigns commented:
> Hey, I been quite busy so I haven't had chance to review it yet, sorry about it. I will get try to review the PR asap.

No worries, take your time :)

> Also, there doesn't seem to be default.json among edited files. Did you run the project after adding records to the localization dictionary? Because it should have exported new records to the said file (and git should have catched it).

Oh, I was not running the exe in the Release folder (oops) since my zomboid server files were not there.
I ran it in the Release folder now and see the default.json file showing in git and just committed it to this branch.

I moved the zomboid server files to the Release folder so that shouldn't happen in the future :)

egebilecen commented on localization/default.json:
```diff
@@ -49,15 +49,21 @@
+  "disc_cmd_set_restart_schedule_type_ok": "Updated the server restart schedule type to {type}.",
   "disc_cmd_set_restart_interval_int_warn": "Restart interval must be at least 60 minutes.",
   "disc_cmd_set_restart_interval_int_ok": "Server restart schedule is updated.",
+  "disc_cmd_set_restart_time_warn": "{time} is an invalid time.",
```
Could you add information about how time format should be?

egebilecen commented on src/Bot/Scheduler.cs:
```diff
@@ -104,4 +104,61 @@ public static void Dispose()
+            DateTime timeDT;
+            try 
+	        {	        
+		        timeDT = DateTime.Parse(time);
```
Some people might try to type time in `"HH.mm"` format. I didn't see any check for catching it. And when it is parsed, it will be interpreted as day and month so it should be checked. Probably in the discord command function.

DomicidalDesigns replied:
I changed this to use ParseExact to only accept "HH:mm" (24-hour format) times

DomicidalDesigns commented:
Thank you for reviewing! 😃
**Changes**
- Time string format only accepts HH:mm
- Added check for no time(s) parameter provided
- Added time format description to command warning and description

egebilecen commented:
Everything is looking good, perfect. Thank you!
